### PR TITLE
[Gst-va] add vpp de-interlace support

### PR DESCRIPTION
**Pull request by FocusLuo**
1, core lib support for gst-va vpp de-interlace
2, test implement support

### CODE REVIEW

**Comment by FocusLuo** [comment]
Session ended. 2 successful, 3 skipped, 4 failed, 0 erroneous. Total duration: 00:00:02
>    assert self.avgrange[0][0] <= self.actual[-3] <= self.avgrange[0][1]
F    AssertionError: assert 0.092 <= 0.07

File: lib/gstreamer/va/util.py [hunk 1]
```diff
@@ -44,6 +44,15 @@ def map_best_hw_format(format, hwformats):
     match_best_format(
       format, set(hwformats) & set(get_supported_format_map().keys())))
 
+@memoize
+def map_deinterlace_method(method):
+  return {
+    "bob"               : "bob",
+    "motion-adaptive"   : "adaptive",
+    "motion-compensated": "compensated",
+    "none"              : "none"
```
Comment (uartie) [inline lib/gstreamer/va/util.py:53]: vadeinterlace does not support `none` value. remove.

File: lib/gstreamer/va/vpp.py [hunk 2]
```diff
@@ -79,9 +79,12 @@ def gen_vpp_opts(self):
 
   def gen_output_opts(self):
     vpp_crop_filter = " videocrop left={left} right={right} top={top} bottom={bottom} ! "
+    vpp_di_filter   = ' vadeinterlace method={mmethod} ! ' if self.hwdevice in ['renderD128' , '0'] else f"va{self.hwdevice}postproc"
```
Comment (uartie) [inline lib/gstreamer/va/vpp.py:82]:
```python
else f"va{self.hwdevice}deinterlace"
```

File: test/gst-va/vpp/deinterlace.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    self.caps = platform.get_caps(
+      "vpp", "deinterlace", self.method.replace('-', '_'))
+
+    if self.caps is None:
+      slash.skip_test(
+        format_value(
+          "{platform}.{driver}.{method} not supported", **vars(self)))
+
+    # The rate is fixed in vaapipostproc deinterlace.  It always outputs at
+    # field rate (one frame of output for each field).
```
Comment (uartie) [inline test/gst-va/vpp/deinterlace.py:51]: Please update this comment... "vaapipostproc" is not part of this context.

**Review by uartie — CHANGES_REQUESTED** [review]

**Comment by uartie** [comment]
> @uartie Windows use type = "nrmse", avg_range = [(0, 0.15)]*3; Linux default used 0.07. Currently gst-va got result is ~0.099. How do you think the nrmse's impact? Do you have any suggestion?

I think 0.10 is reasonable.  The output looks fine on my end.

File: test/gst-va/vpp/deinterlace.py (hunk 3 above)
Comment (FocusLuo) [reply test/gst-va/vpp/deinterlace.py]: updated

File: lib/gstreamer/va/vpp.py (hunk 2 above)
Comment (FocusLuo) [reply lib/gstreamer/va/vpp.py]: updated

File: lib/gstreamer/va/util.py (hunk 1 above)
Comment (FocusLuo) [reply lib/gstreamer/va/util.py]: updated

File: lib/gstreamer/va/vpp.py [hunk 4]
```diff
@@ -79,9 +79,12 @@ def gen_vpp_opts(self):
 
   def gen_output_opts(self):
     vpp_crop_filter = " videocrop left={left} right={right} top={top} bottom={bottom} ! "
+    vpp_di_filter   = ' vadeinterlace method={mmethod} ! ' if self.hwdevice in ['renderD128' , '0'] else f"va{self.hwdevice}deinterlace"
```
[the remark this answers is not in the document]
Comment (FocusLuo) [reply lib/gstreamer/va/vpp.py]: thanks capture this, updated

**Comment by FocusLuo** [comment]
> > @uartie Windows use type = "nrmse", avg_range = [(0, 0.15)]*3; Linux default used 0.07. Currently gst-va got result is ~0.099. How do you think the nrmse's impact? Do you have any suggestion?
> 
> I think 0.10 is reasonable. The output looks fine on my end.

Submitted patch for sanity and bat, not sure that solution is good for you, please help to review and comments

**Comment by uartie** [comment]
> > > @uartie Windows use type = "nrmse", avg_range = [(0, 0.15)]*3; Linux default used 0.07. Currently gst-va got result is ~0.099. How do you think the nrmse's impact? Do you have any suggestion?
> > 
> > 
> > I think 0.10 is reasonable. The output looks fine on my end.
> 
> Submitted patch for sanity and bat, not sure that solution is good for you, please help to review and comments

Thanks.  Use 0.10 for all linux components is easier.

**Comment by FocusLuo** [comment]
> > > > @uartie Windows use type = "nrmse", avg_range = [(0, 0.15)]*3; Linux default used 0.07. Currently gst-va got result is ~0.099. How do you think the nrmse's impact? Do you have any suggestion?
> > > 
> > > 
> > > I think 0.10 is reasonable. The output looks fine on my end.
> > 
> > 
> > Submitted patch for sanity and bat, not sure that solution is good for you, please help to review and comments
> 
> Thanks. Use 0.10 for all linux components is easier.

Sure, will update them in case part

**Comment by FocusLuo** [comment]
> > > > @uartie Windows use type = "nrmse", avg_range = [(0, 0.15)]*3; Linux default used 0.07. Currently gst-va got result is ~0.099. How do you think the nrmse's impact? Do you have any suggestion?
> > > 
> > > 
> > > I think 0.10 is reasonable. The output looks fine on my end.
> > 
> > 
> > Submitted patch for sanity and bat, not sure that solution is good for you, please help to review and comments
> 
> Thanks. Use 0.10 for all linux components is easier.

updated in case part

**Pull request merged**